modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/ExpressionFactoryImpl.java

```diff
@@ -746,6 +735,25 @@ public T get() {
         }
     }
 
+    private static Map.Entry<List<RexNode>, RelDataType> shrinkBounds(IgniteTypeFactory factory, List<RexNode> bound, RelDataType rowType) {
```

Comment:
TBH, I don't fully understand this `shrinkBounds` thing... Let's add tests for `ExpressionFactoryImpl#ranges`. That way it will be easier to verify and debug. Besides, it would be nice to describe how expand bound is supposed work, what invariants it relies on, etc